recipe-server/client/control/components/recipes/AbstractFormPage.js

```diff
@@ -0,0 +1,77 @@
+import autobind from 'autobind-decorator';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+@autobind
+export default class AbstractFormPage extends React.PureComponent {
+  static propTypes = {
+    recipeAction: PropTypes.func.isRequired,
+  };
+
+  state = {
+    formErrors: undefined,
```

Comment:
I played a little with overriding `setState` to prevent subclasses from changing `formErrors`, but that felt a little hacky. I've simply put a comment in `AbstractFormPage` mentioning that while the state _is_ availabe, `formErrors` should not touched.

Reply:
I don't think I explained my concern correctly. Consider this subclass (nonsensical as it is):

```js
class StatefullFormPage extends AbstractFormPage {
  state = {
    subclassState: 42,
  }
  // ... required methods ...
}
```

Since this class sets its own state in the class definition, it would erase the state from AbstractFormPage, right? The same issue applies if the state is set in the constructor. I guess it doesn't matter in this case since the value is `undefined` either way.

Reply:
Ah! Good point. Interesting point. I will investigate.